backend/codesirius/problems/models/execution_constraint.py

```
@@ -0,0 +1,80 @@
+                )
+        elif self.language not in self.problem.languages.all():
+            # this is a create operation
+            raise ValidationError(
+                {"language": "Language is not supported by the problem."}
+            )
+
+    def save(self, *args, **kwargs):
+        self.clean()
+        super().save(*args, **kwargs)
+
+    def __str__(self):
```

Ruff (0.8.2)

Missing return type annotation for special method `__str__`

Add return type annotation: `str`

(ANN204)